src/test/java/mate/academy/security/jwt/JwtTokenProviderTest.java

```diff
@@ -0,0 +1,93 @@
+package mate.academy.security.jwt;
+
+import io.jsonwebtoken.Jwts;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import mate.academy.model.Role;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+
+class JwtTokenProviderTest {
+    private static final String EMAIL = "bob@i.ua";
+    private static final String PASSWORD = "12345678";
+    private static final long VALIDITY_IN_MILLISECONDS = 3600000L;
+    private static final String SECRET = "secret";
+    private JwtTokenProvider jwtTokenProvider;
+    private UserDetailsService userDetailsService;
+    private String token;
+    private List<String> roleList;
+
+    @BeforeEach
+    void setUp() throws ReflectiveOperationException {
+        userDetailsService = Mockito.mock(UserDetailsService.class);
+        jwtTokenProvider = new JwtTokenProvider(userDetailsService);
+
+        Field secretKey = JwtTokenProvider.class.getDeclaredField("secretKey");
```

Comment:
Using reflection to set private fields is not recommended for testing. Consider making the 'secretKey' and 'validityInMilliseconds' fields available to set via constructor.